### DISCUSSION TITLE
gperfools - inclusion of a patch until upstream fixes it

### DISCUSSION THREAD

**Discussion by natale-p** [Homebrew/discussions#2488] [Tap maintenance and Homebrew development]
Hello,

with Monterey, tcmalloc (provided in the gperftool package) stopped working. The relevant link is https://github.com/gperftools/gperftools/issues/1312, in which a simple program:
```
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char *argv[])
{
    void *ptr = malloc(128);
    printf("malloc(128) returned %p\n",ptr);
    free(ptr);
    return 0;
}
```
Segfault when linked against tcmalloc (in the formula of gperftool, it is a similar program, except it uses `tc_*` versions of malloc and free, which never fails). The patch is provided here https://github.com/gperftools/gperftools/pull/1315.

Can we please have this patch in the formula until upstream releases a new version? (it seems the project is dead, the maintainer is not answering nor reviewing PRs).
Thank you!

**Comment by carlocab** [comment]
Thanks for the report. I've opened Homebrew/homebrew-core#89637. Next time, please open a pull request directly.
